xml/src/main/java/org/ehcache/xml/ConfigurationParser.java

```diff
@@ -105,6 +111,22 @@
   private final Map<URI, CacheResourceConfigurationParser> resourceXmlParsers = new HashMap<URI, CacheResourceConfigurationParser>();
   private final ConfigType config;
 
+  static String getReplacementString(String originalValue) {
+    Matcher matcher = SYSPROP.matcher(originalValue);
+
+    String resolvedValue = null;
+    while (matcher.find()) {
+      final String property = matcher.group(1);
+      final String value = System.getProperty(property);
+      if (value == null) {
+        return null;
+      }
+      resolvedValue = matcher.replaceFirst(value);
+      matcher = SYSPROP.matcher(resolvedValue);
```

Comment:
This feels wrong, there ought to be a better way... I just don't know it. (`replaceFirst` just screws around with the matcher's state)